libraries/bsg_manycore.cpp

```diff
@@ -156,56 +156,63 @@ int  hb_mc_manycore_init(hb_mc_manycore_t *mc, const char *name, hb_mc_manycore_
         // Initialize the underlying machine
         if ((err = hb_mc_platform_init(mc, id)) != HB_MC_SUCCESS){
                 free((void*)mc->name);
+                bsg_pr_err("%s: Failed to initialize platform\n", __func__);
                 return err;
         }
 
         // read configuration
         if ((err = hb_mc_manycore_init_config(mc)) != HB_MC_SUCCESS){
                 free((void*)mc->name);
                 hb_mc_platform_cleanup(mc);
+                bsg_pr_err("%s: Failed to initialize config\n", __func__);
                 return err;
         }
 
         // Initialize EVA Maps
         if ((err = hb_mc_manycore_eva_init(mc)) != HB_MC_SUCCESS){
                 free((void*)mc->name);
                 hb_mc_platform_cleanup(mc);
+                bsg_pr_err("%s: Failed to initialize eva maps\n", __func__);
                 return err;
         }
 
         // initialize responders
         if ((err = hb_mc_responders_init(mc))){
                 hb_mc_platform_cleanup(mc);
                 free((void*)mc->name);
+                bsg_pr_err("%s: Failed to initialize responders\n", __func__);
                 return err;
         }
 
         // wait for reset to complete
         if ((err = hb_mc_platform_wait_reset_done(mc)) != HB_MC_SUCCESS) {
                 hb_mc_platform_cleanup(mc);
                 free((void*)mc->name);
+                bsg_pr_err("%s: Failed to wait for reset to complete\n", __func__);
                 return err;
         }
 
         // enable dram
         if ((err = hb_mc_manycore_enable_dram(mc)) != HB_MC_SUCCESS){
                 hb_mc_platform_cleanup(mc);
                 free((void*)mc->name);
+                bsg_pr_err("%s: Failed to enable dram\n", __func__);
                 return err;
         }
 
         // initialize vcaches
         if ((err = hb_mc_manycore_vcache_init(mc)) != HB_MC_SUCCESS) {
                 hb_mc_platform_cleanup(mc);
                 free((void*)mc->name);
+                bsg_pr_err("%s: Failed to initialize vcaches\n", __func__);
                 return err;
         }
 
         // initialize dma
-        if ((err = hb_mc_dma_init(mc)) != HB_MC_SUCCESS) {
-                hb_mc_platform_cleanup(mc);
-                free((void*)mc->name);
-                return err;
+        err = hb_mc_dma_init(mc);
+        if (err != HB_MC_SUCCESS) {
+                bsg_pr_warn("%s: Failed to initialize dma\n", __func__);
+                mc->config.enable_dma = 0;
```

Comment:
If I understand correctly, the intention is that in case dma_init failed, instead of exiting the program, it now downgrades to non-dma mode automatically?

From my understanding, `hb_mc_dma_init` is not only for dma init (this function has a bad name), non-dma test mem also uses this function to init memsys. It may not be okay to move on without exiting the program when error occurs. Afterall, with the new `hb_mc_dma_init_pod_XxYy_hbm` added all pod configurations should have been covered, there is no normal reason for the dma init to fail..

If you really want to downgrade automatically, as mentioned in the other comment it would be better to downgrade in the `bsg_manycore_dma.cpp`, just like what we are already doing for the TESTMEM.

Reply:
I think I'll just remove the change? You make a good point that there's no normal reason for DMA initialization to fail now.